### ISSUE TITLE
Create a CLI frontend to the backend

### ISSUE THREAD

**Issue by rcjsuen** [rcjsuen/dockerfile-language-server-nodejs#79]
While many of the features of the language server protocol doesn't make sense outside of the context of the editor, formatting and validation is something that can performed on a file. We should create a CLI to format and validate Dockerfiles.

**Comment by rcjsuen** [comment]
I've reverted this fix and reopend this because I feel it doesn't make sense for this feature to be available in the language server.

What should be done instead is for all of the heavy lifting to be moved to a new and separate project in its own npm module. That npm module would have its own separate CLI for formatting and validating Dockerfiles.

The language server would then depend on that module and use that module to process and parse the Dockerfile. Whatever results the module returns would be massaged by the language server and then sent back to the client.

**Comment by rcjsuen** [comment]
The first phase of this refactoring has started with the parser being extracted out into its own Git repository at [rcjsuen/dockerfile-ast](https://github.com/rcjsuen/dockerfile-ast).

**Comment by rcjsuen** [comment]
The second phase of this refactoring has completed with the formatter and linter being moved off to a new Git repository at [rcjsuen/dockerfile-utils](https://github.com/rcjsuen/dockerfile-utils).

Feel free to install it with `npm install -g dockerfile-utils` and give it a try if you are interested! :)

**Comment by rcjsuen** [comment]
I've removed the validator and replaced it with a dependency on `dockerfile-utils` in commit cc4d893d128d1d12373a60c063e4ee3531dacf05 but I wrote the wrong issue number in the commit message. The formatter has been partially replaced but most of the code is still around to handle `textDocument/onTypeFormatting` and `textDocument/rangeFormatting` requests. The full document `textDocument/formatting` request however has been modified to delegate to `dockerfile-utils`.

As the language server's validator and full document formatter has been modified to use `dockerfile-utils`, I'm going to consider this particular issue resolved. New issues will be opened for further refactorings as necessary and requests and bugs related to the CLI linter or formatter can be opened in the [rcjsuen/dockerfile-utils repository](https://github.com/rcjsuen/dockerfile-utils).
